### DISCUSSION TITLE
layer wms

### DISCUSSION THREAD

**Discussion by Pueablo** [visgl/deck.gl#7653] [Q&A]
hello! I'm new to deck, before I used leaflet for map layers, but currently I'm trying with deck and I couldn't connect geoserver and deck.
I have searched and commented that getTileData() helps in these cases, but I have not been successful.

I understand that my problem could be the configuration of the url, could you help me with what I am wrong?

My url: 
http://localhost:8080/geoserver/ejemplo/wms?service=WMS&version=1.1.0&request=GetMap&layers=ejemplo%3Aarg_admbndl_ALL_unhcr2017&bbox=-8189395.2664%2C-7374204.276700001%2C-5970884.653999999%2C-2485268.0469000004&width=348&height=768&srs=EPSG%3A3857&styles=&format=application/openlayers

My code: 
`new TileLayer({
            id: 'Map_2',
            getTileData(tile) {
            const {bbox} =  tile;
            const {east, north, south, west} = bbox;
               const urlQueryStringParams = {
                    bbox: [west, south, east, north].join(','),
                    format: 'application/openlayers',
                    height: 768,
                    layers: ['ejemplo', 'arg_admbndl_ALL_unhcr2017'].join(','),
                    request: 'GetMap',
                    service: 'WMS',
                    
                    srs: 'EPSG:3857',
                    styles: '',
                    version: '1.1.0',
                    width: 348
                }
                const urlQueryString = Object.keys(urlQueryStringParams).map(key => key + '=' + urlQueryStringParams[key]).join('&');
                const url = 'http://localhost:8080/geoserver/ejemplo/wms';
              
                return load(url + '?' + urlQueryString)
              },
              
        }),`

And my error in console:
![image](https://user-images.githubusercontent.com/82724510/219513085-490f430d-9d29-41a6-b6fd-2a06dadb30a7.png)

thanks!!

I am following this example: https://github.com/visgl/deck.gl/discussions/6885#discussioncomment-2703052

**Comment by kylebarron** [comment]
https://github.com/visgl/deck.gl/discussions/7636 Might also be helpful

**Reply by ibgreen** [reply]
Can you try that URL directly in the browser URL bar? I can't test it because it references your localhost... 

The error message indicates that you are getting an HTML response instead of an image. If so, that HTML response likely contains an error message explaining what when wrong. You could also look in Chrome's network tab

**Reply by Pueablo** [reply]
Hi, first of all thanks for answering, I can't access that link, do you think it's because of the way the link is posted?

**Reply by Pueablo** [reply]
> #7636 También podría ser útil

Hola Kyle, Hi, first of all thanks, the problem is the url with xyz, i have the layers with openlayers
